components/core/src/clp/telemetry/telemetry.cpp

```
@@ -0,0 +1,76 @@
+
+    opentelemetry::metrics::Provider::SetMeterProvider(std::move(provider));
+}
+
+void shutdown() {
+    auto provider = opentelemetry::metrics::Provider::GetMeterProvider();
+    if (provider) {
+        opentelemetry::metrics::Provider::SetMeterProvider(nullptr);
+    }
+}
+
+void record_query_metrics(uint64_t bytes_scanned, uint64_t bytes_output) {
```

Cppcheck (2.20.0)

[style] The function 'record_query_metrics' is never used.

(unusedFunction)